extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/devconsole/Invocation.java

```diff
@@ -108,7 +109,8 @@ public String getPackageName(String name) {
         private long start;
         private long duration;
         private Method method;
-        private List<Builder> children;
+        // If async processing and the request context is not propagated a new child can be added when/after the builder is built
+        private final List<Builder> children = new CopyOnWriteArrayList<>();
```

Comment:
Wouldn't it make more sense to have a volatile boolean called `built` or similar that is set to true at the very beginning of `build()` and exit from `addChild()` if it's true?

Reply:
I think that `volatile` would not be enough in this case, we would have to synchronize the access anyway. `CopyOnWriteArrayList` should be safe although not so efficient, but that should not be a big problem in the dev mode.